imageregistry/v1/types.go

```diff
@@ -268,6 +268,32 @@ type ImageRegistryConfigStorageAzure struct {
 	CloudName string `json:"cloudName,omitempty"`
 }
 
+// ImageRegistryConfigStorageIBMCOS holds the information to configure
+// the registry to use IBM Cloud Object Storage for backend storage.
+type ImageRegistryConfigStorageIBMCOS struct {
+	// bucket is the bucket name in which you want to store the registry's
+	// data.
+	// Optional, will be generated if not provided.
+	// +optional
+	Bucket string `json:"bucket,omitempty"`
+	// location is the IBM Cloud location in which your bucket exists.
+	// Optional, will be set based on the installed IBM Cloud location.
+	// +optional
+	Location string `json:"location,omitempty"`
+	// resourceGroupName is the name of the IBM Cloud resource group that this
+	// bucket is associated with.
+	// +optional
+	ResourceGroupName string `json:"resourceGroupName,omitempty"`
+	// resourceKeyCrn is the CRN of the IBM Cloud resource key that is created
+	// for the service instance. Required to contain HMAC type credentials.
+	// +optional
+	ResourceKeyCRN string `json:"resourceKeyCrn,omitempty"`
+	// serviceInstanceCrn is the CRN of the IBM Cloud Object Storage service
+	// instance that this bucket is associated with.
+	// +optional
+	ServiceInstanceCRN string `json:"serviceInstanceCrn,omitempty"`
```

Comment:
Explain why this is optional. What happens if this is or isn't defined?

Reply:
The image registry operator will create a new service instance and populate this field with the returned CRN.

Reply:
Small nit pick, but to align with other provider fields casing, we should make this `serviceInstanceCRN` (and `resourceKeyCRN`